cfpq_pyalgo/pygraphblas/tests/test_matrix_all_pairs_reachability.py

```diff
@@ -0,0 +1,63 @@
+import cfpq_pyalgo.pygraphblas as algo
+
+import networkx as nx
+from pyformlang.cfg import CFG
+
+
+def test_empty_graph():
+    g = nx.MultiDiGraph()
+    cfg = CFG.from_text("S -> a")
+
+    reachable_vertices = algo.matrix_all_pairs_reachability(g, cfg)
+    assert reachable_vertices == []
+
+
+def test_no_reachable_vertices():
+    g = nx.MultiDiGraph()
+    g.add_edge(0, 1, label="b")
+    g.add_edge(1, 2, label="c")
+    g.add_edge(2, 0, label="d")
+
+    cfg = CFG.from_text("S -> S S | a")
+
+    reachable_vertices = algo.matrix_all_pairs_reachability(g, cfg)
+    assert reachable_vertices == []
+
+
+def test_worst_case():
+    # The graph is two cycles of coprime lengths with a single common vertex.
+    # The first cycle is labeled by the open bracket and the second cycle is labeled by the close bracket
+    g = nx.MultiDiGraph()
+    g.add_edge(0, 1, label="a")
```

Comment:
Is it true that vertices should be numbered from 0 to n-1? If yes, then it should be described in the appropriate class or method.

Reply:
I added conversion of vertices numbers between NetworkX MultiDiGraph and BooleanMatrixGraph